Libraries/LibGUI/GFilePicker.h

```diff
@@ -7,22 +7,41 @@ class GLabel;
 
 class GFilePicker final : public GDialog {
 public:
-    GFilePicker(const StringView& path = "/", CObject* parent = nullptr);
+    enum class Mode {
+        Open,
+        Save
+    };
+
+    static String get_open_filepath();
+    static String get_save_filepath();
+
+    GFilePicker(Mode type = Mode::Open, const StringView& path = "/", CObject* parent = nullptr);
     virtual ~GFilePicker() override;
 
     FileSystemPath selected_file() const { return m_selected_file; }
 
-    virtual const char* class_name() const override { return "GFilePicker"; }
-
 private:
     void set_preview(const FileSystemPath&);
     void clear_preview();
 
-    GTableView* m_view { nullptr };
+    const String ok_button_name()
+    {
+        switch (m_mode) {
+        case Mode::Open:
+            return "Open";
+        case Mode::Save:
+            return "Save";
+        default:
+            return "OK";
+        }
+    }
+
+    GTableView* m_view{ nullptr };
     NonnullRefPtr<GDirectoryModel> m_model;
     FileSystemPath m_selected_file;
 
-    GLabel* m_preview_image_label { nullptr };
-    GLabel* m_preview_name_label { nullptr };
-    GLabel* m_preview_geometry_label { nullptr };
+    GLabel* m_preview_image_label{ nullptr };
```

Comment:
There should be a space before the `{` here.
Are you using an old `clang-format`? Please use version 7 or newer as the older versions are a bit buggy.